### PR TITLE
Completed Testing in community.py resolves issue #6184

### PR DESCRIPTION
**Pull request by Mjh9122**
Adds testing coverage in community.py. Only branch not tested is the return statement of a nested function, which I do not know how to test. Couple of best practice questions. 
1. When testing internal functions, is it OK to simply call them directly, or should they be covered exclusively by forcing another function to call them? 
2. When forcing an ImportError, what is the best practice, I used sys.modules, but am unsure of other (better) ways.
Let me know of any other issues
#6184

### CODE REVIEW

File: networkx/generators/tests/test_community.py [hunk 1]
```diff
@@ -251,18 +272,99 @@ def test_mu_too_small():
...
+    with pytest.raises(
+        nx.ExceededMaxIterations, match="Could not match average_degree"
+    ):
+        nx.generators.community._generate_min_degree(2, 2, 1, 0.01, 0)
+
+
+# Simply testing that no error is raise when scipy.special is not available
+def test_generate_min_degree_no_sp():
+    sys.modules["scipy.special"] = None
+    assert 1 == nx.generators.community._generate_min_degree(2, 2, 1, 100.0, 10)
```
Comment (MridulS) [inline networkx/generators/tests/test_community.py:370]: There is no need to test this honestly. There isn't a nice/easy way to test these import errors.

File: networkx/generators/tests/test_community.py [hunk 2]
```diff
@@ -133,6 +139,9 @@ def test_gaussian_random_partition_graph():
     pytest.raises(
         nx.NetworkXError, nx.gaussian_random_partition_graph, 100, 101, 10, 1, 0
     )
+    # Test when clusters are likely less than 1
+    G = nx.gaussian_random_partition_graph(10, 0.5, 0.5, 0.5, 0.5, seed=1)
+    assert len(G) == 10
```
Comment (MridulS) [inline networkx/generators/tests/test_community.py:144]: Hmm, in this test we aren't really testing the output of `gaussian_random_partition_graph`, as the length of nodes will always be input number of nodes. The other asserts here in `test_gaussian_random_partition_graph` also don't cover the functionality IMO. Here we are just testing the code is hitting all branches not if it's correct.

But we don't need to fix that in this PR, just writing some thoughts down :)

**Review by MridulS — CHANGES_REQUESTED** [review]
Thanks for this @Mjh9122!

I think this is ready to go, could you just remove the scipy.special test?

**Comment by Mjh9122** [comment]
@MridulS Just took it out, thanks for your help

File: networkx/generators/tests/test_community.py (hunk 2 above)
Comment (Mjh9122) [reply networkx/generators/tests/test_community.py:142]: Yeah I thought as much, but just figured I'd follow what was already there. From my understanding of the function, the edges are random, so a test on the size of the graph is not particularly helpful. Maybe a test with the probabilities set to 0 or 1 could help verify edge cases, but idk how to test a more general use case.

Comment (rossbar) [reply networkx/generators/tests/test_community.py:142]: Agreed - this would be a nice follow-up!

**Review by rossbar — APPROVED** [review]
LGTM, thanks @Mjh9122 !

**Review by MridulS — APPROVED** [review]
